src/diffusers/configuration_utils.py

```diff
@@ -162,6 +170,21 @@ def save_config(self, save_directory: Union[str, os.PathLike], push_to_hub: bool
         self.to_json_file(output_config_file)
         logger.info(f"Configuration saved in {output_config_file}")
 
+        # Additionally, save the implementation file too. It can happen for a pipeline, for a model, and
+        # for a scheduler.
+        if self.__class__.__name__ not in _all_available_pipeline_component_modules:
```

Comment:
```suggestion
        if self.__class__.__name__ not in _all_available_pipelines_schedulers_model_classes:
```

Reply:
Actually, I think this is too dangerous and we shouldn't touch it as we do now. What if people fork diffusers and then create a custom class? This custom class will then be saved as its own file

Reply:
There's too much manual intervention otherwise, I think. You need to ensure modules are placed in certain locations and it can get complicated for pipelines like SDXL that have many components. Then you have to edit model index manually, which is easy to miss. 

To make it safer, we could expose a function argument `save_custom_modules` and default it to False. That way, it won't be saved by default. Users can set it to True as they see fit. 

But also, this path won't be hit always and I don't see how it could create the problem you are thinking of.